Assignment_Week_3.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace MICP_Assignments
+{
+    public class Assignment_Week_3
+    {
+        public static int[][] uniqueTripletCombinations(int[] userInput)
+        {
+            //checks to ensure that there IS input into the array
+            if (userInput == null || userInput.Length <= 0)
+                return null;
+
+            //if there isn't enough input
+            if (userInput.Length < 3)
+                throw new ArgumentException();
+
+            //used to contain the pending results
+            List<int[]> results = new List<int[]>();
+
+            //organizes the user input
+            Array.Sort(userInput);
+
+            //O(n^3) Complexity, this REALLY does need to be improved

```

Comment:
An idea to improve your solution may be to implement some sort of memoization :blush:, for example, you could implement a set to know if you have already checked a number. Also, twoSum could be good place to start.